{{cookiecutter.project_slug}}/clients/web/vue3/src/services/base-model.js

```diff
@@ -0,0 +1,8 @@
+import { z } from 'zod'
+import {readonly} from '@thinknimble/tn-models-fp'
+
```

Comment:
@lakardion getting a zod error here that id is required when using userApi.create(), I temporarily set it to `.optional` to get it working, but I thought that by making it readonly it would not be an issue?

Reply:
sounds like a bug. but at the same time you need not to make id readonly. it is readonly by default

Reply:
Tracking here. As a workaround, not marking `id` as readonly it should work!
https://github.com/thinknimble/tn-models-fp/issues/132

Reply:
okay will leave this as is for now

Reply:
I think there's a bug with the readonly implementation when used on `id`, due to its duality with being readonly regardless being marked as readonly or not. 
readonly should be fine for other fields, until I get to the aforementioned issue we should probably leave `id` without the `readonly` wrap

Reply:
Okay will do, regardless the issue was occurring before I added the readonly, I was just trying to fix it by using the readonly method

Reply:
Okay I didn't notice the create shape override. Yeah readonly is not supported in the override models. Will fix that so that they're completely ignored.

Reply:
Upgrade to `2.12.3` and lmk how it goes

https://github.com/thinknimble/tn-models-fp/pull/137